lib/plexy/logger.ex

```diff
@@ -97,16 +98,31 @@ defmodule Plexy.Logger do
   end
 
   defp metric_name(metric, name) do
-    app = Plexy.Config.get(:plexy, :app_name, System.get_env("APP_NAME") || "plexy")
     name = to_string(name)
-    "#{name}##{app}.#{metric}"
+    "#{name}##{app_name()}.#{metric}"
+  end
+
+  defp app_name do
+    @plexy_config.get(:plexy, :app_name) || raise "You must set app_name for Plexy config"
```

Comment:
This should use `Plexy.Config` it is a nice interface that has abstractions around setting runtime values and defaults.

Reply:
it _does_ use it, except in test where I stub the value

Reply:
in the `config/test.exs` just set the name.

instead of
```elixir
config :plexy, config: Plexy.TestConfig
```

do this
```elixir
config :plexy, app_name: "test-app-name"
```
no need for all this dependency injection code

or you can just set it in a `setup\0` block

```elixir
setup do
  Application.set_env(:plexy, :app_name, "test-app-name")
end
```